### PR TITLE
[WIP][FEATURE] Added json-schema-validation to core

### PR DESCRIPTION
**Pull request by turbo-ele**

### CODE REVIEW

**Comment by chirimoya** [comment]
Tests

File: src/Sulu/Bundle/CoreBundle/Resources/config/listener.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <parameters>
+        <parameter key="sulu_core.controller_listener.class">Sulu\Component\Listener\ControllerListener</parameter>
+    </parameters>
+
+    <services>
+        <service id="sulu_core.controller_listener" class="%sulu_core.controller_listener.class%">
+            <argument type="service" id="annotation_reader"/>
+            <tag name="kernel.event_listener" event="kernel.controller" method="onKernelController" />
```
Comment (chirimoya) [inline src/Sulu/Bundle/CoreBundle/Resources/config/listener.xml]: Should this listener alway be enabled?

File: src/Sulu/Bundle/CoreBundle/Listener/ControllerListener.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+     */
+    public function onKernelController(FilterControllerEvent $event)
+    {
+        $controller = $event->getController();
+
+        list($object, $method) = $controller;
+
+        // the controller could be a proxy, e.g. when using the JMSSecuriyExtraBundle or JMSDiExtraBundle
+        $className = ClassUtils::getClass($object);
+        $reflectionClass = new \ReflectionClass($className);
```
Comment (chirimoya) [inline src/Sulu/Bundle/CoreBundle/Listener/ControllerListener.php]: I'm personally not a big fan of Reflection. Is there another possibility to solve this problem?

File: src/Sulu/Bundle/CoreBundle/Listener/ControllerListener.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace PoolAlpin\Bundle\BaseBundle\Listener;
```
Comment (dantleech) [inline src/Sulu/Bundle/CoreBundle/Listener/ControllerListener.php]: I think this won't work :)

**Comment by dantleech** [comment]
What is the purpose of this feature?

**Comment by turbo-ele** [comment]
Once it's done, it will add the possibility of validating API's by a ruleset, defined in a json file.

**Comment by wachterjohannes** [comment]
@turbo-ele ping

**Comment by webmozart** [comment]
Closing due to lack of feedback.

**Pull request closed without merging**